File: examples/nodejs/backup_file_zero_bytes.js
#!/usr/bin/env node
/**
 * Backup File Zero Bytes
 * Problem: Backup file is created but is zero bytes
 * Solution: Send file size in payload, validate in DeadManPing panel
 * Documentation: https://deadmanping.com/backup-file-zero-bytes
 */

const fs = require('fs');
const { execSync } = require('child_process');
const https = require('https');

const MONITOR_ID = 'YOUR_MONITOR_ID';  // Replace with your actual monitor ID
const backupFile = '/backups/db-backup.sql.gz';

// Run backup
execSync('pg_dump mydb | gzip > ' + backupFile);

// Get file size
let fileSize = 0;
try {
  fileSize = fs.statSync(backupFile).size;
} catch (e) {
  // File doesn't exist
}

// Always ping with file size in payload
// In DeadManPing panel: set validation rule "size" > 0
// Panel will automatically detect if size is 0
https.request(`https://deadmanping.com/api/ping/${MONITOR_ID}?size=${fileSize}`, { method: 'POST' }).end();

File: examples/python/backup_file_zero_bytes.py
#!/usr/bin/env python3
"""
Backup File Zero Bytes
Problem: Backup file is created but is zero bytes
Solution: Send file size in payload, validate in DeadManPing panel
Documentation: https://deadmanping.com/backup-file-zero-bytes
"""

import os
import subprocess
import requests

MONITOR_ID = "YOUR_MONITOR_ID"  # Replace with your actual monitor ID
backup_file = "/backups/db-backup.sql.gz"

# Run backup
subprocess.run(["pg_dump", "mydb"], stdout=open(backup_file.replace('.gz', ''), "w"))
subprocess.run(["gzip", backup_file.replace('.gz', '')])

# Get file size
file_size = os.path.getsize(backup_file) if os.path.exists(backup_file) else 0

# Always ping with file size in payload
# In DeadManPing panel: set validation rule "size" > 0
# Panel will automatically detect if size is 0
requests.post(f"https://deadmanping.com/api/ping/{MONITOR_ID}?size={file_size}")
